lib/rubygems/util.rb

```diff
@@ -116,4 +116,18 @@ def self.traverse_parents directory, &block
     end
   end
 
+  ##
+  # Globs for files matching +pattern+ inside of +directory+,
+  # returning absolute paths to the matching files.
+
+  if RUBY_VERSION >= "2.5"
```

Comment:
It's better to inject this condition into `self.glob_files_in_dir` method.

Reply:
But then we incur the cost of the condition every time the method is called

Reply:
Can you show the benchmark about method call? I understand your opinion. But I didn't see that its cost is better than code style.

Reply:
I personally agree with @hsbt here.

Reply:
I suspect the cost of the condition will always be far less that the cost of reading a dir, SSD or not...

Reply:
Conditional in the method seems to be about 85% as fast as pre-computing it

Reply:
Turning around, are conditionally defined methods really that bad?